### ISSUE TITLE
block-opening-brace-space-before false positive in SugarSS

### ISSUE THREAD

**Issue by Arcanemagus** [stylelint/stylelint#2001]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug where `block-opening-brace-space-before` triggers on SugarSS files (that don't have `{`).

> Which rule, if any, is this issue related to?

`block-opening-brace-space-before`

> What CSS is needed to reproduce this issue?

```sugarss
.selector
  box-shadow: 1px 0 9px rgba(0, 0, 0, .4)
```

> What stylelint configuration is needed to reproduce this issue?

`stylelint-config-standard`

> Which version of stylelint are you using?

`7.5.0`, this is a regression from `7.4.2`, possibly due to #1958?

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Node API (`linter-stylelint` Atom package).

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to SugarSS.

> What did you expect to happen?

I expected a warning about the missing leading zero to be the only message.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
spec/fixtures/sugarss/bad.sss
 1:9   ×  Expected single space before "{"   block-opening-brace-space-before
 2:39  ×  Expected a leading zero            number-leading-zero
```

Is this a bug in the SugarSS generation, or does a note need to be added [here](https://github.com/stylelint/stylelint-config-standard#using-the-config-with-sugarss-syntax) about needing to disable this rule, like there is for `declaration-block-trailing-semicolon`?

**Comment by evilebottnawi** [comment]
@Arcanemagus i think need disable, in sugarss we don't have `braces`, let's wait what says @jeddy3 or @davidtheclark

**Comment by davidtheclark** [comment]
I agree that since we already have that caveat listed, we should just add another caveat and expect users to disable the rule. Unless someone wants to dig in a little and see if there's a simple solution.

**Comment by jeddy3** [comment]
> we should just add another caveat and expect users to disable the rule

Agreed. Done in https://github.com/stylelint/stylelint-config-standard/pull/46

**Comment by Arcanemagus** [comment]
I'll add that to the list of automated exceptions when editing SugarSS files then, closing this issue and thanks for the decision!
